### ISSUE TITLE
esModule option is ignored

### ISSUE THREAD

**Issue by nicolashenry** [systemjs/builder#775]
The systemjs esModule option (https://github.com/systemjs/systemjs/blob/master/docs/config-api.md) seems to be ignored with systemjs-builder, is it supposed to only work with systemjs alone ?

**Comment by aluanhaddad** [comment]
Yes it should work with the builder. What does your configuration look?

**Comment by nicolashenry** [comment]
I made a minimal example here : https://github.com/nicolashenry/esmodule-systemjs-builder

"index.html" display "ok" in console
"index-bundled.html" display "enableProdMode is not a function" (the same happen if I remove 'esModule: true' with index.html)

Maybe I'm doing something wrong here.

**Comment by aluanhaddad** [comment]
Unless something has changed, module format meta config should look like this.

```js
meta: {
  '@angular/core/*': {
      format: 'esm'
    }
}
```

**Comment by nicolashenry** [comment]
I don't understand, I use the "esModule" option and not the "format" option.

**Comment by nicolashenry** [comment]
My objective is to use a legacy module "bundles/core.umd.js" which use "named imports", "bundles/core.umd.js" is not an ecmascript module.

**Comment by wongwill86** [comment]
Is the esModule flag working? React doesn't export __esModule right now and I would like to do `import * as React from 'react'` but I can't seem to get the correct incantation for this property
Here's a very simple project that imports react and importing redux for comparison.
https://github.com/wongwill86/esmoduleissue

**Comment by guybedford** [comment]
@nicolashenry the bundled version still needs the configuration file to set the `esModule` meta configuration.

**Comment by wongwill86** [comment]
Is there someone else other than https://github.com/nicolashenry/esmodule-systemjs-builder/blob/master/systemjs.config.js#L26 where the meta should be set?

**Comment by nicolashenry** [comment]
@guybedford buildStatic is supposed to "To make a bundle that is independent of the SystemJS loader entirely" so I don't understand why and how I could add the ```esModule``` meta configuration if I don't have any systemjs runtime.

**Comment by guybedford** [comment]
Ahh, SFX support for the `esModule` flag is not currently included.

This can be added by enabling an `esModule` flag input for our build transformers in https://github.com/systemjs/babel-plugin-transform-system-register, https://github.com/systemjs/babel-plugin-transform-cjs-system-wrapper, https://github.com/systemjs/babel-plugin-transform-global-system-wrapper, https://github.com/systemjs/babel-plugin-transform-amd-system-wrapper, which adds a:

```javascript
System.register([..], function (require, exports, module) {
  // this line is added by the flag:
  Object.defineProperty(exports, '__esModule', {  value: true });
});
```

If anyone is interested in working on this please let me know, otherwise I will try get to it when I can.

**Comment by guybedford** [comment]
I've included the updates here in for the coming release https://github.com/systemjs/builder/commit/802ab434308a2dd7e568c30f2d7f82fdd6571727.

**Comment by guybedford** [comment]
Released in 0.16.5.
